clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -107,6 +103,23 @@ public void stringSerdeShouldSupportDifferentEncodings() {
         }
     }
 
+    @Test
+    public void listSerdeShouldRoundtripInput() {
+        List<Integer> testData = Arrays.asList(1, 2, 3);
+        Serde<List<Integer>> listSerde = Serdes.ListSerde(ArrayList.class, Serdes.Integer());
+        assertEquals("Should get the original " + List.class +
+                        " after serialization and deserialization", testData,
+                listSerde.deserializer().deserialize(topic, listSerde.serializer().serialize(topic, testData)));
+    }
+
+    @Test
+    public void listSerdeShouldReturnNull() {
+        List<Integer> testData = Arrays.asList();
+        Serde<List<Integer>> listSerde = Serdes.ListSerde(ArrayList.class, Serdes.Integer());
+        assertEquals("Should get null after serialization and deserialization on an empty list", null,
```

Comment:
I think the List serde should return null (after a round trip) if and only if the input was null. If the input was an empty list, then the list serde should instead return an empty list.